src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -6355,6 +6408,416 @@ register_handler(int sig, struct sigaction *old_handler)
 	}
 }
 
+/* free all EQs allocated for aio */
+
+static void
+free_aio_ctx(void)
+{
+	int i;
+
+	for (i = 0; i < aio_eq_count; i++) {
+		daos_eq_destroy(aio_eq_list[i].eq, 0);
+		D_MUTEX_DESTROY(&aio_eq_list[i].lock);
+	}
+	aio_eq_count = 0;
+}
+
+/* aio functions return negative errno in case of failure */
+static int
+create_ev_eq_for_aio(aio_ctx_t *aio_ctx);
+int
+io_setup(int maxevents, io_context_t *ctxp)
+{
+	int        rc;
+	aio_ctx_t *aio_ctx_obj;
+
+	if (next_io_setup == NULL) {
+		next_io_setup = dlsym(RTLD_NEXT, "io_setup");
+		D_ASSERT(next_io_setup != NULL);
+	}
+
+	rc = next_io_setup(maxevents, ctxp);
+	if (rc < 0)
+		return rc;
+
+	D_ALLOC(aio_ctx_obj, sizeof(aio_ctx_t));
+	if (aio_ctx_obj == NULL) {
+		io_destroy(*ctxp);
+		return -ENOMEM;
+	}
+	aio_ctx_obj->ctx   = *ctxp;
+	aio_ctx_obj->depth = maxevents;
+	atomic_store_relaxed(&aio_ctx_obj->num_op_submitted, 0);
+	atomic_store_relaxed(&aio_ctx_obj->num_op_done, 0);
```

Comment:
nit: There's an atomic_init() function which would be clearer here.

Reply:
Now they are replaced by atomic_init(). Thank you!